internal/util.go

```diff
@@ -71,12 +71,17 @@ func GetResourceLabel(resource client.Object, labelName string) (string, error)
 	return labelValue, nil
 }
 
-func DefaultCacheOptions() cache.Options {
+func DefaultCacheOptions(namespaces []string) cache.Options {
+	defaultNamespaces := map[string]cache.Config{}
+	for _, v := range namespaces {
+		defaultNamespaces[v] = cache.Config{}
+	}
 	return cache.Options{
 		ByObject: map[client.Object]cache.ByObject{
 			&apicorev1.Secret{}: {
 				Label: k8slabels.Everything(),
 			},
 		},
+		DefaultNamespaces: defaultNamespaces,
```

Comment:
Do we really need to define the namespace? What happens if we keep like before? If it's just for performance gain, I  don't think it is worth the effort to introduce such options to offer the namespace to be cached. It increase the complex to config LM.

Reply:
No it has to be there because otherwise it requires the role bindings to be cluster scope not namespace